### PR TITLE
feat(caas): backport from caas

### PR DESCRIPTION
**Pull request by haarchri**
### Description of your changes
- backport from [caas](https://github.com/upbound/configuration-caas/tree/main)
- update 
  - XAKS
     - add region as input
     - add providerConfig as input
  - XCluster
     - add region as input
     - add providerConfig as input
     - using patchSets
  - XNetwork
  - XService 
    - add flux
    - use prometheus as default
- rename folder package -> apis
- enable usages for XAKS, XServices and XApp
- switch crossplane.yaml to root folder
- update Makefile 
- bump submodule

Fixes #

I have:

- [x] Read and followed Upbound's [contribution process](https://git.io/fj2m9).
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Added `backport release-x.y` labels to auto-backport this PR, as appropriate.

### How has this code been tested
uptest

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -57,9 +60,11 @@ build.init: $(UP)
 
 # This target requires the following environment variables to be set:
 # - UPTEST_CLOUD_CREDENTIALS, cloud credentials for the provider being tested, e.g. export UPTEST_CLOUD_CREDENTIALS=$(cat azure.json)
+# - To ensure the proper functioning of the end-to-end test resource pre-deletion hook, it is crucial to arrange your resources appropriately. 
+#   You can check the basic implementation here: https://github.com/upbound/uptest/blob/main/internal/templates/01-delete.yaml.tmpl.
```
Comment (ytsarev) [inline Makefile:64]:
```suggestion
#   You can check the basic implementation here: https://github.com/upbound/uptest/blob/main/internal/templates/03-delete.yaml.tmpl
```
it already new place in main.

File: Makefile [hunk 2]
```diff
@@ -57,9 +60,11 @@ build.init: $(UP)
 
 # This target requires the following environment variables to be set:
 # - UPTEST_CLOUD_CREDENTIALS, cloud credentials for the provider being tested, e.g. export UPTEST_CLOUD_CREDENTIALS=$(cat azure.json)
+# - To ensure the proper functioning of the end-to-end test resource pre-deletion hook, it is crucial to arrange your resources appropriately.
```
Comment (ytsarev) [inline Makefile:63]: As we are not using test hooks anymore, is it still valid statement?

File: crossplane.yaml [hunk 3]
```diff
@@ -36,15 +36,17 @@ spec:
...
-    - provider: xpkg.upbound.io/upbound/provider-azure-containerservice
-      version: ">=v0.33.0"
-    - provider: xpkg.upbound.io/upbound/provider-azure-dbformariadb
-      version: ">=v0.33.0"
-    - provider: xpkg.upbound.io/upbound/provider-azure-dbforpostgresql
-      version: ">=v0.33.0"
-    - provider: xpkg.upbound.io/upbound/provider-azure-network
-      version: ">=v0.33.0"
-    - provider: xpkg.upbound.io/crossplane-contrib/provider-helm
-      version: ">=v0.15.0"
```
Comment (ytsarev) [inline crossplane.yaml:50]: It looks like we are changing the indentation style for arrays in this whole configuration. it does not look consistent with https://github.com/upbound/platform-ref-aws/blob/main/crossplane.yaml#L32
Can we avoid reindentation where it is not needed?

**Review by ytsarev — COMMENTED** [review]
Added the initial feedback

File: package/cluster/composition.yaml
Comment (ytsarev) [inline package/cluster/composition.yaml:1]: Can we make `git mv package/cluster/composition.yaml apis/cluster/composition.yaml` so the actual diff will be visible?

Comment (haarchri) [reply package/cluster/composition.yaml:1]: i will split this PR for better visibility: first: https://github.com/upbound/platform-ref-azure/pull/43

**Comment by haarchri** [comment]
followup via: https://github.com/upbound/platform-ref-azure/pull/44

**Pull request closed without merging**